stdlib/parray.mc

```diff
@@ -0,0 +1,369 @@
+  match pa with PAData(3,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(4,x0,x1,x2,y ,x4,x5,x6,x7,x8,x9) else
+  match pa with PAData(4,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(5,x0,x1,x2,x3,y ,x5,x6,x7,x8,x9) else
+  match pa with PAData(5,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(6,x0,x1,x2,x3,x4,y ,x6,x7,x8,x9) else
+  match pa with PAData(6,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(7,x0,x1,x2,x3,x4,x5,y ,x7,x8,x9) else
+  match pa with PAData(7,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(8,x0,x1,x2,x3,x4,x5,x6,y ,x8,x9) else
+  match pa with PAData(8,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9)
+           then PAData(9,x0,x1,x2,x3,x4,x5,x6,x7,y ,x9) else
+  match pa with PAData(9,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9) then
+           if eqi l 0 then PANode(1,10,PAData(
+                   10,x0,x1,x2,x3,x4,x5,x6,x7,x8,y),
+                   empty,empty,empty,empty,empty,empty,empty,empty,empty)
+           else
+            PANext(PAData(10,x0,x1,x2,x3,x4,x5,x6,x7,x8,y)) else
+  match pa with PANode(i,l2,x0,x1,x2,x3,x4,x5,x6,x7,x8,x9) in
+    let e = (match i with 0 then x0 else match i with 1 then x1 else
+             match i with 2 then x2 else match i with 3 then x3 else
+             match i with 4 then x4 else match i with 5 then x5 else
+             match i with 6 then x6 else match i with 7 then x7 else
+             match i with 8 then x8 else x9) in
+    let e2 = workAddPA e y (divi l2 10) in
+    let setNode = lam e. lam i. lam i2.
+      match i with 0 then PANode(i2,l2,e,x1,x2,x3,x4,x5,x6,x7,x8,x9) else
+      match i with 1 then PANode(i2,l2,x0,e,x2,x3,x4,x5,x6,x7,x8,x9) else
+      match i with 2 then PANode(i2,l2,x0,x1,e,x3,x4,x5,x6,x7,x8,x9) else
+      match i with 3 then PANode(i2,l2,x0,x1,x2,e,x4,x5,x6,x7,x8,x9) else
+      match i with 4 then PANode(i2,l2,x0,x1,x2,x3,e,x5,x6,x7,x8,x9) else
+      match i with 5 then PANode(i2,l2,x0,x1,x2,x3,x4,e,x6,x7,x8,x9) else
+      match i with 6 then PANode(i2,l2,x0,x1,x2,x3,x4,x5,e,x7,x8,x9) else
+      match i with 7 then PANode(i2,l2,x0,x1,x2,x3,x4,x5,x6,e,x8,x9) else
+      match i with 8 then PANode(i2,l2,x0,x1,x2,x3,x4,x5,x6,x7,e,x9) else
+                          PANode(i2,l2,x0,x1,x2,x3,x4,x5,x6,x7,x8,e) in
+    match e2 with PANext e3 then
+      if eqi i 9 then
+        if eqi l 0 then
+            PANode(1, muli l2 10,setNode e3 i 10,
+              empty,empty,empty,empty,empty,empty,empty,empty,empty)
+          else PANext (setNode e3 i 10)
+      else setNode e3 i (addi i 1)
+    else setNode e2 i i
+end
+
+
+
+-- `addPA pa y` adds an element `y` to the end of the
+-- pure array `pa` and returns a new pure array
+-- Either keep track of the number of elements in your pure array in another
+-- variable, or use `lengthPA` to get the current number of elements.
+let addPA = lam pa. lam y.
+  workAddPA pa y 0
+
+
+
+/-
+recursive
+let addPA : all a. PA a -> a -> PA a = lam pa. lam y.
```

Comment:
Is this a duplicate definition of `addPA`?

Reply:
It is commented-out code that I forgot to remove. It is removed in the next commit.